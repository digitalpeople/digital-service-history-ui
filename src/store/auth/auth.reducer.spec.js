import reducer from './auth.reducer';
import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT,
} from '../action.type';

describe('auth.reducer', () => {
  describe('initial state', () => {
    describe('unmanaged action type', () => {
      it('should return state', () => {
        const action = {
          type: 'test-action-type',
        };

        const actual = reducer(undefined, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_REQUEST, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_REQUEST,
        };

        const actual = reducer(undefined, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_SUCCESS, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_SUCCESS,
          userId: 'test-user-id',
        };

        const actual = reducer(undefined, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_FAILURE, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_FAILURE,
          error: {
            mesage: 'test-error-message',
          },
        };

        const actual = reducer(undefined, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGOUT, () => {
      it('should return state', () => {
        const action = {
          type: LOGOUT,
        };

        const actual = reducer(undefined, action);

        expect(actual).toMatchSnapshot();
      });
    });
  });
  describe('modified state', () => {
    const state = {
      foo: 'bar',
    };

    describe('unmanaged action type', () => {
      it('should return state', () => {
        const action = {
          type: 'test-action-type',
        };

        const actual = reducer(state, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_REQUEST, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_REQUEST,
        };

        const actual = reducer(state, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_SUCCESS, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_SUCCESS,
          userId: 'test-user-id',
        };

        const actual = reducer(state, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGIN_FAILURE, () => {
      it('should return state', () => {
        const action = {
          type: LOGIN_FAILURE,
          error: {
            mesage: 'test-error-message',
          },
        };

        const actual = reducer(state, action);

        expect(actual).toMatchSnapshot();
      });
    });
    describe(LOGOUT, () => {
      it('should return state', () => {
        const action = {
          type: LOGOUT,
        };

        const actual = reducer(state, action);

        expect(actual).toMatchSnapshot();
      });
    });
  });
});
