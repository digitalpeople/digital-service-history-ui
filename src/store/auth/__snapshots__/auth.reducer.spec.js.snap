// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth.reducer initial state @auth/LOGIN_FAILURE should return state 1`] = `
Object {
  "isAuthenticated": false,
  "message": undefined,
  "userId": null,
}
`;

exports[`auth.reducer initial state @auth/LOGIN_REQUEST should return state 1`] = `
Object {
  "isAuthenticated": false,
  "message": null,
  "userId": null,
}
`;

exports[`auth.reducer initial state @auth/LOGIN_SUCCESS should return state 1`] = `
Object {
  "isAuthenticated": true,
  "message": null,
  "userId": "test-user-id",
}
`;

exports[`auth.reducer initial state @auth/LOGOUT should return state 1`] = `
Object {
  "isAuthenticated": false,
  "message": null,
  "userId": null,
}
`;

exports[`auth.reducer initial state unmanaged action type should return state 1`] = `
Object {
  "isAuthenticated": false,
  "message": null,
  "userId": null,
}
`;

exports[`auth.reducer modified state @auth/LOGIN_FAILURE should return state 1`] = `
Object {
  "foo": "bar",
  "message": undefined,
}
`;

exports[`auth.reducer modified state @auth/LOGIN_REQUEST should return state 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`auth.reducer modified state @auth/LOGIN_SUCCESS should return state 1`] = `
Object {
  "foo": "bar",
  "isAuthenticated": true,
  "userId": "test-user-id",
}
`;

exports[`auth.reducer modified state @auth/LOGOUT should return state 1`] = `
Object {
  "isAuthenticated": false,
  "message": null,
  "userId": null,
}
`;

exports[`auth.reducer modified state unmanaged action type should return state 1`] = `
Object {
  "foo": "bar",
}
`;
